source/acquaman/OLDSGM/SGMXASScanController.cpp

```diff
@@ -0,0 +1,294 @@
+	else
+		tmpBAction = SGMBeamline::sgm()->createSDD1EnableAction(false);
+	tmpBAction ? initializationActions_->appendAction(0, tmpBAction) : initializationFailed = true;
+	if(config_->allDetectorConfigurations().isActiveNamed(SGMBeamline::sgm()->amptekSDD2()->detectorName()))
+		tmpBAction = SGMBeamline::sgm()->createSDD2EnableAction(true);
+	else
+		tmpBAction = SGMBeamline::sgm()->createSDD2EnableAction(false);
+	tmpBAction ? initializationActions_->appendAction(0, tmpBAction) : initializationFailed = true;
+	*/
+
+//	tmpBAction = SGMBeamline::sgm()->synchronizedDwellTime()->createMasterTimeAction(config_->regionTime(0));
+	tmpBAction ? initializationActions_->appendAction(0, tmpBAction) : initializationFailed = true;
+	tmpAction = new AMBeamlineControlMoveAction(SGMBeamline::sgm()->nextDwellTimeTrigger());
+	tmpAction->setSetpoint(0);
+	tmpAction ? initializationActions_->appendAction(0, tmpAction) : initializationFailed = true;
+	tmpAction = new AMBeamlineControlMoveAction(SGMBeamline::sgm()->nextDwellTimeConfirmed());
+	tmpAction->setSetpoint(1);
+	tmpAction ? initializationActions_->appendAction(0, tmpAction) : initializationFailed = true;
+
+	tmpBAction = SGMBeamline::sgm()->scaler()->createStartAction(0);
+	tmpBAction ? initializationActions_->appendAction(0, tmpBAction) : initializationFailed = true;
+	tmpBAction = SGMBeamline::sgm()->scaler()->createContinuousEnableAction(false);
+	tmpBAction ? initializationActions_->appendAction(0, tmpBAction) : initializationFailed = true;
+
+	initializationActions_->appendStage(new QList<AMBeamlineActionItem*>());
+	tmpBAction = SGMBeamline::sgm()->scaler()->createScansPerBufferAction(1);
+	tmpBAction ? initializationActions_->appendAction(1, tmpBAction) : initializationFailed = true;
+	tmpBAction = SGMBeamline::sgm()->scaler()->createTotalScansAction(1);
+	tmpBAction ? initializationActions_->appendAction(1, tmpBAction) : initializationFailed = true;
+
+	initializationActions_->appendStage(new QList<AMBeamlineActionItem*>());
+
+	tmpAction = new AMBeamlineControlMoveAction(SGMBeamline::sgm()->energy());
+	tmpAction->setSetpoint(config_->startEnergy());
+	initializationActions_->appendAction(2, tmpAction);
+
+	initializationActions_->appendStage(new QList<AMBeamlineActionItem*>());
+	AMBeamlineControlSetMoveAction* tmpSetAction = new AMBeamlineControlSetMoveAction(SGMBeamline::sgm()->trackingSet());
+	tmpSetAction->setSetpoint(config_->trackingGroup());
+	initializationActions_->appendAction(3, tmpSetAction);
+
+	tmpAction = new AMBeamlineControlMoveAction(SGMBeamline::sgm()->harmonic());
+	tmpAction->setSetpoint(config_->fluxResolutionGroup().controlNamed(SGMBeamline::sgm()->harmonic()->name()).value());
+	initializationActions_->appendAction(3, tmpAction);
+
+	initializationActions_->appendStage(new QList<AMBeamlineActionItem*>());
+	initializationActions_->appendAction(4, SGMBeamline::sgm()->createBeamOnActions());
+
+	beamlineInitialized_ = true;
```

Comment:
same question here. it looks like we will always return true for this function.

We didn't use `cleanupFailed` as well.